### ISSUE TITLE
HSQLDB support

### ISSUE THREAD

**Issue by binkley** [JetBrains/Exposed#450]
Perhaps it is a documentation issue of the frontpage README, but I'd like to see HSQLDB explicitly supported.

**Comment by Tapac** [comment]
No, there is no dialect for HSQLDB in Exposed, but you can implement it and provide us with PR.

**Comment by bgmf** [comment]
I tried to do that, but struggle to cope with the wildly distributed APIs and whatnot. Currently I try to remove the quotes from the query, that renders SQL like `SELECT ROLES."id", ROLES."name" FROM ROLES` and results in `user lacks privilege or object not found: ROLES.id`.

If you use a DB tool of your choice, you can recreate the issue by using a query like stated. If you now remove the quotes, it works.

But I have no idea, how to tell the system, to not us quotes (at least not in the given three default classes `VendorDialect`, `DataTypeProvider` and `FunctionProvider`). Do I need to explicitly implement the parent interface `DatabaseDialect`?

Oh and yes, I registered my dialect with the `Database` class...

**Comment by dirk0108** [comment]
Is there any news on this? Are there any plans to officially support HslqDB?

**Comment by bgmf** [comment]
I wouldn't bet on it. And to be honest: I don't see the need anymore. With Docker etc. you can just spin up a supported local DB fast...

**Comment by dirk0108** [comment]
HsqlDB is used in many applications in embedded operation. This is also the case in my production application. A Docker container can't help me with that ;)
